### DISCUSSION TITLE
Disabling Chrome Autofill Issue in Text Box

### DISCUSSION THREAD

**Discussion by SanjayKalaivanan** [community/community#65469] [Programming Help] [answered]
I'm currently facing an issue with Chrome's autofill feature in text boxes. Despite setting autocomplete="off" and trying various approaches, Chrome continues to suggest and apply values that are unrelated. I'd greatly appreciate any suggestions on overcoming this challenge.

![autofill](https://github.com/community/community/assets/95475582/17c4b3f5-55c0-4f4c-bcb8-4b4cd42302c5)

**Comment by talha4real** [comment, marked as answer]
The autocomplete="off" attribute is often ignored by modern browsers for certain fields, such as username and password fields, due to security reasons. However, if you're dealing with other types of fields and still encountering this issue, there are several workarounds you can try:

1. Use a Unique name and id Attribute
Sometimes using a unique name for the name and id attributes can prevent the browser from recognizing the field and thereby disable autofill.

<input type="text" name="unique_name_12345" id="unique_id_12345" />



2. Use autocomplete="new-password"
For password fields, you can trick the browser into thinking it's a new password field, which it won't try to autofill.



<input type="password" autocomplete="new-password" />



3. Use Hidden Dummy Fields
Place hidden fields before the actual fields. Browsers often fill the first set of fields they find.


<input type="text" name="fake_username" style="display:none;" />
<input type="password" name="fake_password" style="display:none;" />




4. Change Field Type Dynamically
Use JavaScript to change the input field type immediately after the page loads. This will usually prevent the browser from recognizing it as an autofill field.


window.onload = function() {
  const myInput = document.getElementById('my_input');
  myInput.type = 'text';
}



5. Delay Form Rendering
If you are using a front-end framework, you could delay the rendering of the input fields until after the page has loaded, which can also prevent autofill in some cases.



6. Use JavaScript to Clear Fields on Page Load
As a last resort, you can use JavaScript to clear the fields as soon as the page loads.


window.onload = function() {
  document.getElementById('my_input').value = '';
}
